source/cursorManager.py

```diff
@@ -145,6 +145,10 @@ def _caretMovementScriptHelper(
 		extraDetail=False,
 		handleSymbols=False,
 	):
+		if isScriptWaiting():
+			# Moving / reporting is quite costly, so we don't want to do it if there are more scripts waiting.
+			# Otherwise, NVDA could become unusable while it processes many backed up scripts.
+			return
```

Comment:
When returning early, we bypass all the rest of this function. May this have side effects?

More specifically I ask such a question du to the following comment:
```
# #10343: Speak before setting selection because setting selection might
# move the focus, which might mutate the document, potentially invalidating
# info if it is offset-based.
```

and at the end of the function:
`self.selection = selection`

Reply:
I think the only side affect is that you may move down the page slower than before. I.e. the selection will now only move when there is no other script waiting. That comment is only to do with state within the function, but since  the body of the function is now not running sometimes, it is not really relevant. Still, if you notice any odd behaviour with this pr, please report.

Reply:
OK. Thanks for the explanation.

I have not thouroughly tested it. But the tests I have done in Word and Outlook are successful and the freeze does not occur anymore.